client/maisie/resources/models.py

```diff
@@ -93,17 +96,27 @@ def download(self, id: int):
             request = request.json()
             if (
                 ("data") in request
+                and "checksum" in request["data"]
                 and "_links" in request["data"]
                 and "name" in request["data"]
                 and "download" in request["data"]["_links"]
             ):
                 download_link = request["data"]["_links"]["download"]
                 download_data = session.get(download_link)
                 model_name = request["data"]["name"]
-        if path:
+                download_checksum = request["data"]["checksum"]
+        local_checksum = md5(download_data.content).hexdigest()
+        if path and model_name:
             model_name = os.path.join(path, model_name)
-        with open(model_name, "wb") as model_file:
-            model_file.write(download_data.content)
+        response = "Checksums differ"
+        if local_checksum and download_checksum and local_checksum == download_checksum:
+            with open(model_name, "wb") as model_file:
+                model_file.write(download_data.content)
```

Comment:
Shouldn't we first save the file and then calculate the hash of an *already* locally saved file?

I am aware it is a minor issue, but verifying checksums after writing the contents of `download_data.content` to the download destination would allow us to write the file in chunks and still check its integrity afterwards.

Now that I think about this, `download_data` should probably also be renamed to something more self-explanatory.